### PR TITLE
Pathway analysis: Meta-analysis with QuSAGE

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
#33 - meta-analysis with QuSAGE

In this notebook, I use two medulloblastoma datasets—[Northcott et al.](https://doi.org/10.1038/nature11327) and [Robinson et al.](https://doi.org/10.1038/nature11213)—to demonstrate how to identify pathways that are differentially expressed or activated in between the SHH subgroup and Groups 3 and 4 in both datasets using QuSAGE.

### CODE REVIEW

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+
+}
+```
+
+### Gene sets
+
+`qusage` allows you to read in gene sets that are in the [GMT format](http://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29).
+[MSigDB](http://software.broadinstitute.org/gsea/msigdb) offers genesets in this format.
+
+For more information or recommendations about gene sets, [see the Gene Sets
```
Comment (cansav09) [inline pathway-analysis/qusage_meta_analysis.Rmd]:
```suggestion
For more information or recommendations about gene sets, [see the 'Gene Sets'
```
Maybe something to make more clear it's a headline and that's why you are capitalizing it? IDK

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+
+The Quantitative Set Analysis of Gene Expression (QuSAGE) 
+([Yaari et al. _NAR_. 2013.](https://doi.org/10.1093/nar/gkt660)) framework
+has advantages that we outline in 
+[`qusage_single_dataset`](./qusage_single_dataset.nb.html), including the 
+fact that it returns more than just a p-value.
+Specifically, QuSAGE quantifies gene set activity with a full probability 
+density function (PDF). 
+If we're interested in pathway analysis of multiple datasets, QuSAGE allows
+us to perform a _meta-analysis_ by combining distributions from the QuSAGE
```
Comment (cansav09) [inline pathway-analysis/qusage_meta_analysis.Rmd]: Why is this in italics?

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+```{r}
+combined_results <- combinePDFs(results_list)
+```
+
+Just as in a single dataset, we can extract relevant information from 
+the combined `QSArray` with built-in `qusage` functions.
+
+Let's look at the top 20 pathways with `qsTable`.
+
```
Comment (cansav09) [inline pathway-analysis/qusage_meta_analysis.Rmd]: Maybe you could add a note here or down later about the use of these path indices and how to find them. Or just to pull attention to that they are in this table and we will use them again down below.

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+First let's plot KEGG Pathways in Cancer, which has elevated expression in
+the SHH group as it has a _positive_ fold change.
+
+```{r}
+plotCombinedPDF(combined_results, path.index = 162)
+legend("topleft", legend = c("Northcott", "Robinson", "Meta-analysis"),
+       lty = 1, col = c("#E41A1C", "#377EB8", "black"))
+```
+
+Let's plot KEGG Dorsoventral Axis Formation, which has higher expression in
```
Comment (cansav09) [inline pathway-analysis/qusage_meta_analysis.Rmd]:
```suggestion
Let's plot KEGG Dorsoventral Axis Formation, which has higher expression in
```

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+
+#### Plot individual gene mean and 95% confidence intervals
+
+We know that the directionality of the KEGG Dorsoventral Axis Formation pathway
+agrees between datasets.
+We can look into what genes are driving the pathway activity with the 
+`plotCIsGenes` function.
+
+```{r}
+plotCIsGenes(northcott_results, path.index = 107)
```
Comment (cansav09) [inline pathway-analysis/qusage_meta_analysis.Rmd]: Can you use `cex.xaxis` again and see if the labels can be a tad larger?

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+legend("topleft", legend = c("Northcott", "Robinson", "Meta-analysis"),
+       lty = 1, col = c("#E41A1C", "#377EB8", "black"))
+```
+
+#### Plot individual gene mean and 95% confidence intervals
+
+We know that the directionality of the KEGG Dorsoventral Axis Formation pathway
+agrees between datasets.
+We can look into what genes are driving the pathway activity with the 
+`plotCIsGenes` function.
```
Comment (cansav09) [inline pathway-analysis/qusage_meta_analysis.Rmd]: In these plots, can you explain what the horizontal dashed blue line represents and also what the significance of the darkness of the vertical lines is?

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+
+```{r}
+qsTable(combined_results)
+```
+
+### Plotting
+
+We can plot the distributions with `plotCombinedPDF`.
+First let's plot KEGG Pathways in Cancer, which has elevated expression in
+the SHH group as it has a _positive_ fold change.
```
Comment (cansav09) [inline pathway-analysis/qusage_meta_analysis.Rmd]: Can you put a quick one sentence summary about what the x-axis of "Pathway Activity" represents?

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 8]
```diff
@@ -0,0 +1,429 @@
...
+Looks like some of the top genes are the same between datasets: `56907`, `56776`
+Because Entrez IDs are not particularly human-readable, we can use the same
+annotation package to convert these to gene symbols and gene names.
+
+```{r}
+AnnotationDbi::select(org.Hs.eg.db, keys = c("56907", "56776"), 
+                      keytype = "ENTREZID", columns = c("SYMBOL", "GENENAME"))
+```
+
+It looks like these two genes promote the polymerization of actin filaments.
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply pathway-analysis/qusage_meta_analysis.Rmd]: Good call

File: pathway-analysis/qusage_meta_analysis.Rmd (hunk 6 above)
Comment (jaclyn-taroni) [reply pathway-analysis/qusage_meta_analysis.Rmd]: Used the option to drop the vertical lines -- don't think the shade means anything

File: pathway-analysis/qusage_meta_analysis.Rmd (hunk 1 above)
Comment (jaclyn-taroni) [reply pathway-analysis/qusage_meta_analysis.Rmd]: Made it bold

**Pull request merged**